app/js/services/impactStudyService.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+angular.module('ddsApp').service('ImpactStudyService', function($location, $http, $sessionStorage, $log, uuid) {
+    var RESEARCH_DOMAIN = 'https://mes-droits.fr',  // this domain is owned by economy researchers mandated by Pole Emploi and DGCS
+        STUDIED_PRESTATIONS = [ 'aah', 'aah_non_calculable', 'acs', 'adpa', 'af', 'aide_logement', 'aide_logement_non_calculable', 'alf', 'als', 'apl', 'asf', 'asi', 'aspa', 'ass', 'bourse_college', 'bourse_lycee', 'cf', 'cmu_c', 'paje_base', 'paris_logement_familles', 'ppa', 'rsa', 'rsa_majore', 'rsa_non_calculable', 'rsa_non_majore' ];
+
+
+    function getSessionId() {
+        $sessionStorage.sessionId = $sessionStorage.sessionId || uuid.v4();
+        return $sessionStorage.sessionId;
+    }
+
+    function getResearchId() {
+        $sessionStorage.researchId = $sessionStorage.researchId || $location.search().eid;
```

Comment:
We actually do! That’s why we try to get it from the storage first.

But… hey, you’re right, that should be local storage! Good catch  :)